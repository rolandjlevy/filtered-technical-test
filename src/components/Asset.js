import React from 'react';

class Asset extends React.Component {
    render() {
        return (
            <React.Fragment>
                <li className={`asset type-${this.props.classNameType}`}>
                    <h2>{this.props.name}</h2>
                    <p>
                        <span className="info">{this.props.duration} minutes</span>
                        <span className="info">{this.props.type}</span>
                    </p>
                    <button className="remove-asset">x</button>
                </li>
            </React.Fragment>
        )
    }
}

export default Asset;