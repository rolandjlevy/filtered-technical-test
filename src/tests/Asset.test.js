import React from 'react';
import Asset from '../components/Asset';

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Component', () => {
  test('renders', () => {
    const li = shallow(<Asset />);
    expect(li).toMatchSnapshot();
  });
});
