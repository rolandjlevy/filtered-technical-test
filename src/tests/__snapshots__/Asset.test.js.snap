// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Asset />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <li
        className="asset type-undefined"
>
        <h2 />
        <p>
                <span
                        className="info"
                >
                         minutes
                </span>
                <span
                        className="info"
                />
        </p>
        <button
                className="remove-asset"
        >
                x
        </button>
</li>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <p>
            <span
                        className="info"
            >
                         minutes
            </span>
            <span
                        className="info"
            />
</p>,
          <button
            className="remove-asset"
>
            x
</button>,
        ],
        "className": "asset type-undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="info"
>
                 minutes
</span>,
              <span
                className="info"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  " minutes",
                ],
                "className": "info",
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " minutes",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "info",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "x",
            "className": "remove-asset",
          },
          "ref": null,
          "rendered": "x",
          "type": "button",
        },
      ],
      "type": "li",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <li
          className="asset type-undefined"
>
          <h2 />
          <p>
                    <span
                              className="info"
                    >
                               minutes
                    </span>
                    <span
                              className="info"
                    />
          </p>
          <button
                    className="remove-asset"
          >
                    x
          </button>
</li>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2 />,
            <p>
              <span
                            className="info"
              >
                             minutes
              </span>
              <span
                            className="info"
              />
</p>,
            <button
              className="remove-asset"
>
              x
</button>,
          ],
          "className": "asset type-undefined",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="info"
>
                   minutes
</span>,
                <span
                  className="info"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    " minutes",
                  ],
                  "className": "info",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  " minutes",
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "info",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "x",
              "className": "remove-asset",
            },
            "ref": null,
            "rendered": "x",
            "type": "button",
          },
        ],
        "type": "li",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
